netty/src/main/java/io/grpc/netty/ProtocolNegotiator.java

```diff
@@ -44,6 +44,9 @@
 
   /**
    * The Netty handler to control the protocol negotiation.
+   *
+   * <p>TODO(carl-mastrangelo): Replace ChannelHandler with ChannelInboundHandler or
```

Comment:
Hmm... ChannelHandler may still be correct. exceptionCaught is only deprecated in ChannelHandler; it's still fine in ChannelInboundHandler. Implementations should maybe extending `ChannelDuplexHandler`